projecttacoma/bulk-export-app
# Display status code and content location of export request after execution

# Summary

This PR adds a page to the app to display the status code and the content location for the bulk export kickoff request
[JIRA Issue](https://jira.mitre.org/browse/TACOSTRAT-1517)

## New behavior

When you navigate from the query builder page to the export-execution page, using the button to the right of the query string, you should see the status code and the content location on a new page for the bulk data export that you built with the query builder. When you click the button on the builder page, it starts the kickoff request, which will either be successful or not successful indicated by a green search icon or a red retry icon respectively.

## Code changes
- `src/app/export-execution/layout.tsx` is the layout for this new export-execution page
- `src/app/export-execution/page.tsx` export execution page that displays the status code of the request and the content location if the request is successful.
- `src/components/query-string.tsx` added the link to the next page 
- `src/util/exportRequestBuilders.ts` made type params a possibly undefined field to be able to correctly get the data from the query string on the export execution page. Also this is probably a slight improvement on this builder function

# Testing guidance
- run `npm run check`
- Check that prettier and eslint cause no errors
- `npm run build`
- `npm run start`
- check that when you build a query and navigate to the execution page that the status code and content location are visible.
- An example of a bad request you can make is if you select group and any id and then ActivityDefinition in the dropdown and try to navigate to the next page using the arrow to the right of the export request string. You should see a red retry button with a tooltip telling you the error.
